WarshipGame/Assets/Scripts/Gameplay/ShipManager.cs

```diff
@@ -0,0 +1,148 @@
+    /// <param name="hexGameObject"></param>
+    public void HandleTerrainSelected(GameObject hexGameObject)
+    {
+        if (SelectedShip == null || !PlayersTurn) return;
+
+        HexData selectedHex = hexGameObject.GetComponent<HexData>();
+
+        if (HandelHexOutOfRange(selectedHex.Grid) || HandleSelectedHexIsShipHex(selectedHex.Grid)) return;
+
+        HandleTargetHexSelected(selectedHex);
+    }
+    
+    /// <summary>
+    /// Clears the Range of the ships if its not selected
+    /// And draws the range if the ship is selected
+    /// </summary>
+    /// <param name="shipReference"></param>
+    private void PrepareShipForMovement(Ship shipReference)
+    {
+        if (SelectedShip != null)
+        {
+            ClearOldSelection();
+        }
+
+        SelectedShip = shipReference;
+        SelectedShip.Select();
+        MovementScript.ShowRange(SelectedShip, HexGridScript);
+    }
+
+    /// <summary>
+    /// Hides the range of the previous turn
+    /// And empties the previouslySelectedHex variable for next turn
+    /// </summary>
+    private void ClearOldSelection()
+    {
+        _previouslySelectedHex = null;
+        SelectedShip.Deselect();
+        MovementScript.HideRange(HexGridScript);
+        SelectedShip = null;
+    }
+
+    /// <summary>
+    /// Shows the path of the selected hex to Ship
+    /// If the hex is pressed again invoke the Ship movement
+    /// </summary>
+    /// <param name="selectedHex"></param>
+    private void HandleTargetHexSelected(HexData selectedHex)
+    {
+        if (_previouslySelectedHex == null || _previouslySelectedHex != selectedHex)
+        {
+            _previouslySelectedHex = selectedHex;
+            MovementScript.ShowPath(selectedHex.Grid, HexGridScript);
+            return;
+        }
+
+        MovementScript.MoveShip(SelectedShip,HexGridScript); 
+        PlayersTurn = false;
+        SelectedShip.MovementFinished += ResetTurn;
+        ClearOldSelection();
+    }
+    
+    /// <summary>
+    /// If you select a ship and you then click the hex the ship is standing on
+    /// this function will prevent an error and just deselect the ship
+    /// </summary>
+    /// <param name="hexPosition"></param>
+    /// <returns></returns>
+    private bool HandleSelectedHexIsShipHex(Vector2Int hexPosition)
+    {
+        if (hexPosition != HexGridScript.GetClosestHex(SelectedShip.transform.position)) return false;
+        
+        SelectedShip.Deselect();
+        ClearOldSelection();
+            
+        return true;
+    }
+    
+    /// <summary>
+    /// If the selected hex is not inside the movement range of the currently selected ship it will just be ignored
+    /// </summary>
+    /// <param name="hexPosition"></param>
+    /// <returns></returns>
+    private bool HandelHexOutOfRange(Vector2Int hexPosition)
+    {
+        if (MovementScript.IsHexInRange(hexPosition)) return false;
+        //Hex out of range, deselect maybe?
+        return true;
```

Comment:
comment is redundant
if can be turned into a return `return !MovementScript.IsHexInRange(hexPosition);`